src/lib/utilities.js

```diff
@@ -351,6 +351,39 @@ module.exports = {
 		};
 	},
 
+	/**
+	 * Generates a filter function to be used with `Array.filter()` when filtering a list of Devices
+	 * by some value. Supports `online`, `offline`, Platform Name, Device ID, or Device Name
+	 *
+	 * @param {string} filter - Filter value to use for filtering a list of devices
+	 * @returns {function|null}
+	 */
+	buildDeviceFilter(filter) {
+		const { knownPlatforms } = module.exports;
+		let filterFunc = null;
+		if (filter){
+			const platforms = knownPlatforms();
+			if (filter === 'online') {
+				filterFunc = (d) => {
+					return d.connected;
+				};
```

Comment:
minor nit - you can collapse these down to the concise form:

```js
filterFunc = (d) => d.connected;
```

arrow fns have an implicit return ([docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#Function_body))

Reply:
Indeed! Good call out. I'm far too comfortable with the function block format and overlook this refactor option _all of the time_.